### PR TITLE
Improve test suite speed

### PR DESCRIPTION
**Pull request by pedro-stanaka**
## Summary

Using `t.Parallel` to speed up tests and also creating a faster make target to test locally (running all tests, but using build caches).
Also, pulling some of data creation on tests outside so we can avoid generating a lot of data without reason.

Having faster test suit improves feedback loop and CI times.

The native histogram tests are still taking an awful amount of time, but I did not had time to take a look into extracting data creation.

On main I get:

```
      329.71 real       411.05 user        37.16 sys
```

On my branch I get:

```
       80.68 real       434.55 user        90.90 sys
```

without racing it is a much better:

```
real    0m13.486s    user    0m48.597s   sys     0m4.594s
```

#### Changes

- **test speedup**
- **Improving test speed by using parallel testing**
- **Adding target to test without race detector**
- **The export in the makefile had no side-effects**

### CODE REVIEW

File: engine/distributed_test.go [hunk 1]
```diff
@@ -255,34 +248,54 @@ func TestDistributedAggregations(t *testing.T) {
...
+					EnableAtModifier:     true,
+				},
+			}
 			for _, test := range tests {
+				test := test
+				var allSeries []*mockSeries
+				remoteEngines := make([]api.RemoteEngine, 0, len(test.seriesSets)+1)
+				for _, s := range test.seriesSets {
+					remoteEngines = append(remoteEngines, engine.NewRemoteEngine(
+						localOpts,
```
Comment (MichaHoffmann) [inline engine/distributed_test.go:268]: before we would have set lookback here; but now we declare the engines with default lookback right?

Comment (pedro-stanaka) [reply engine/distributed_test.go]: You are right, good catch. Pulled data creation for this test again and put engine creation a bit futher down so we can use the new `localOpt` with the proper lookbackDelta.
Cyclomatic complexity on this test is really high.

File: engine/engine_test.go [hunk 2]
```diff
@@ -4740,24 +4759,29 @@ func TestNativeHistograms(t *testing.T) {
 		},
 	}
 
+	defer pprof.StopCPUProfile()
```
Comment (MichaHoffmann) [inline engine/engine_test.go:4762]: this still needed? we could just run with "--cpuprofile test.pprof" instead i think

**Comment by MichaHoffmann** [comment]
Awesome, this is great! Thanks

```
$ time go test .
ok  	github.com/thanos-io/promql-engine/engine	30.370s
```

**Review by MichaHoffmann — APPROVED** [review]
LGTM

File: engine/distributed_test.go (hunk 1 above)
Comment (MichaHoffmann) [reply engine/distributed_test.go]: Yeah, I think at some point we need to clean up our testing story somehow!

**Pull request merged**